meeteval/wer/matching/cy_levenshtein.pyx

```diff
@@ -0,0 +1,279 @@
+# distutils: language = c++
+#cython: language_level=3
+
+import numpy as np
+from libcpp.vector cimport vector
+from libcpp.pair cimport pair
+
+ctypedef unsigned int uint
+
+cdef extern from "levenshtein.h":
+    uint levenshtein_distance_(
+            vector[uint] reference,
+            vector[uint] hypothesis,
+    )
+
+    uint levenshtein_distance_custom_cost_(
+            vector[uint] reference,
+            vector[uint] hypothesis,
+            uint cost_del,
+            uint cost_ins,
+            uint cost_sub,
+            uint cost_cor,
+    )
+
+    uint time_constrained_levenshtein_distance_[T](
+            vector[uint] reference,
+            vector[uint] hypothesis,
+            vector[pair[T, T]] reference_timing,
+            vector[pair[T, T]] hypothesis_timing,
+            uint cost_del,
+            uint cost_ins,
+            uint cost_sub,
+            uint cost_cor,
+    )
+
+    uint time_constrained_levenshtein_distance_unoptimized_[T](
+            vector[uint] reference,
+            vector[uint] hypothesis,
+            vector[pair[T, T]] reference_timing,
+            vector[pair[T, T]] hypothesis_timing,
+            uint cost_del,
+            uint cost_ins,
+            uint cost_sub,
+            uint cost_cor,
+    )
+
+    struct LevenshteinStatistics:
+        uint insertions
+        uint deletions
+        uint substitutions
+        uint correct
+        uint total
+        vector[pair[uint, uint]] alignment
+
+    LevenshteinStatistics time_constrained_levenshtein_distance_with_alignment_[T](
+            vector[uint] reference,
+            vector[uint] hypothesis,
+            vector[pair[T, T]] reference_timing,
+            vector[pair[T, T]] hypothesis_timing,
+            uint cost_del,
+            uint cost_ins,
+            uint cost_sub,
+            uint cost_cor,
+            uint eps
+    )
+
+
+def obj2vec(a, b):
+    int2sym = dict(enumerate(sorted(set(a) | set(b))))
+    sym2int = {v: k for k, v in int2sym.items()}
+    return [sym2int[a_] for a_ in a], [sym2int[b_] for b_ in b]
+
+def levenshtein_distance(
+        reference,
+        hypothesis,
+):
+    reference, hypothesis = obj2vec(reference, hypothesis)
+
+    return levenshtein_distance_(reference, hypothesis)
+
+def levenshtein_distance_custom_cost(
+        reference,
+        hypothesis,
+        cost_del,
+        cost_ins,
+        cost_sub,
+        cost_cor,
+):
```

Comment:
How about merging `levenshtein_distance` and `levenshtein_distance_custom_cost` and add a comment in the code, that `levenshtein_distance_` is faster?